examples/prompting/Basic_Code_Generation.ipynb

```diff
@@ -0,0 +1,310 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bxGr_x3MRA0z"
+      },
+      "source": [
+        "<table class=\"tfo-notebook-buttons\" align=\"left\">\n",
+        "  <td>\n",
+        "    <a target=\"_blank\" href=\"https://colab.research.google.com/github/google-gemini/cookbook/blob/main/examples/prompting/Basic_Code_Generation.ipynb\"><img src = \"https://www.tensorflow.org/images/colab_logo_32px.png\"/>Run in Google Colab</a>\n",
+        "  </td>\n",
+        "</table>"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ysy--KfNRrCq"
+      },
+      "source": [
+        "The Gemini API can be a great tool to save you time during the development process. Tasks such as code generation, debugging, or optimization can be done with the assistance of the Gemini model."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "Ne-3gnXqR0hI"
+      },
+      "outputs": [],
+      "source": [
+        "!pip install -U -q google-generativeai"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "id": "EconMHePQHGw"
+      },
+      "outputs": [],
+      "source": [
+        "import google.generativeai as genai\n",
+        "\n",
+        "from IPython.display import Markdown"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eomJzCa6lb90"
+      },
+      "source": [
+        "## Configure your API key\n",
+        "\n",
+        "To run the following cell, your API key must be stored it in a Colab Secret named `GOOGLE_API_KEY`. If you don't already have an API key, or you're not sure how to create a Colab Secret, see [Authentication](https://github.com/google-gemini/cookbook/blob/main/quickstarts/Authentication.ipynb) for an example."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 4,
+      "metadata": {
+        "id": "v-JZzORUpVR2"
+      },
+      "outputs": [],
+      "source": [
+        "from google.colab import userdata\n",
+        "GOOGLE_API_KEY=userdata.get('GOOGLE_API_KEY')\n",
+        "\n",
+        "genai.configure(api_key=GOOGLE_API_KEY)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "yQnqEPjephXi"
+      },
+      "source": [
+        "## Examples"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bR-OOcC6pIm5"
+      },
+      "source": [
+        "### Error handling"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "kVF8ZQ38Vs1P"
+      },
+      "outputs": [],
+      "source": [
+        "error_handling_system_prompt =f\"\"\"\n",
```

Comment:
The output is painfully long here, explaining a bunch of problems that are not the one I'm having. Can we simplify to the System Instructions to something like: "Your task is to explain exactly why this error occurred and how to fix it."?